### PR TITLE
stop doubling of events inferred by parameters if midicmd is set

### PR DESCRIPTION
**Pull request by yaxu**

### CODE REVIEW

File: classes/DirtEventTypes.sc [hunk 1]
```diff
@@ -92,9 +95,9 @@ DirtEventTypes {
 					});
 				};
 
-				if(~progNum.notNil)   { var num = ~progNum;   schedmidi.value({ midiout.program(chan, num) })};
-				if(~midibend.notNil)  { var val = ~midibend;  schedmidi.value({ midiout.bend(chan, val)    })};
-				if(~miditouch.notNil) { var val = ~miditouch; schedmidi.value({ midiout.touch(chan, val)   })};
+				if(~progNum.notNil)   { var num = ~progNum;   donecmd.value(\program); schedmidi.value({ midiout.program(chan, num) })};
```
Comment (yaxu) [inline classes/DirtEventTypes.sc:98]: I'm curious why this works:
```
donecmd.value(\program); schedmidi.value({ midiout.program(chan, num) })};
```
But this doesn't:
```
schedmidi.value({ midiout.program(chan, num) })}; donecmd.value(\program);
```
The call of `donecmd` seemed to be delayed in the latter case?

Comment (telephon) [reply classes/DirtEventTypes.sc:98]: `schedmidi` is delayed and happens after `donecmd` in both cases.

Comment (yaxu) [reply classes/DirtEventTypes.sc:98]: I meant in the latter example, the command ends up being sent twice.

Comment (yaxu) [reply classes/DirtEventTypes.sc:98]: Anyway it works as-is, I think this is ok for merging

**Comment by telephon** [comment]
Although I think that now you can't send two commands in one event, right? Not sure what was intended. It is slightly messy in terms of functionality. Maybe you could add an explanation in a longer comment in the file?

**Comment by yaxu** [comment]
@telephon Unless I've made a mistake, you can send multiple commands in one event, but only one of each type.

**Comment by yaxu** [comment]
(The purpose of this bit of code was to make sure that a command is only sent once if its 'midicmd' is inferred from its parameters, as well as being set explicitly)

**Comment by telephon** [comment]
Ok, I just don't get what `donecmd` is doing.

**Comment by yaxu** [comment]
It unsets `midicmd` if its action has already been inferred and done.

**Comment by telephon** [comment]
Ah I see. But wouldn't it have been easier to write instead of:
```supercollider
if(~progNum.notNil)   { var num = ~progNum;   donecmd.value(\program); schedmidi.value({ midiout.program(chan, num) })};
```

simply
```supercollider
if(~progNum.notNil or: { midicmd === \program })   { var num = ~progNum;  schedmidi.value({ midiout.program(chan, num) })};
```

**Comment by yaxu** [comment]
But then there's a bit later on that deals with the midicmd variable in a generic way, so it needs to be unset.

**Pull request merged**